src/Rings/MPolyQuo.jl

```diff
@@ -947,3 +947,55 @@ function dim(a::MPolyQuoIdeal)
   return a.dim
 end
 #######################################################
+
+
+################################################################################
+#
+#  Minimalizing a set of subalgebra generators in graded case
+#
+################################################################################
+@doc Markdown.doc"""
+    minimal_subalgebra_generators(V::Vector{S}) where S <: Union{MPolyElem, MPolyQuoElem}
+
+Given a vector `V` collecting homogeneous elements of a graded multivariate polynomial ring or
+quotient ring, say `R`, return a minimal subset of these elements which still generate the
+subalgebra of `R` generated by all elements collected in `V`.
+
+NOTE: The conditions on `V` and `R` are automatically checked.
+
+# Examples
+```jldoctest
+julia> R, (x,y,z) =  GradedPolynomialRing(QQ, ["x", "y", "z"])
+(Multivariate Polynomial Ring in x, y, z over Rational Field graded by
+  x -> [1]
+  y -> [1]
+  z -> [1], MPolyElem_dec{fmpq, fmpq_mpoly}[x, y, z])
+
+julia> V = [x,y,x^2+y^2]
+3-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
+ x
+ y
+ x^2 + y^2
+
+julia> minimal_subalgebra_generators(V)
+2-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
+ y
+ x
+```
+"""
+function minimal_subalgebra_generators(V::Vector{S}) where S <: Union{MPolyElem, MPolyQuoElem}
+  p = parent(V[1])
+  @assert all(x->parent(x) == p, V)
+  p isa MPolyRing_dec && isgraded(p) || throw(ArgumentError("The base ring must be graded"))
+  all(ishomogeneous, V) || throw(ArgumentError("The input data is not homogeneous"))
+
+  # iterate over the generators, starting with those in lowest degree, then work up
+  W = sort(V, by = x -> degree(x)[1])
+  result = [ W[1] ]
+  for elm in W
+    if !subalgebra_membership(elm, result)[1]
+       push!(result, elm)
+    end
+  end
+  return result
```

Comment:
@wdecker I rewrote `minimal_subalgebra_generators`, please have a look to see if you agree it is still correct.

Reply:
@fingolfin: Yes, much better! Thanks, Wolfram

Reply:
@fingolfin: But there is again this doctest error which I thought I had already corrected:
```jldoctest
│ julia> R, (x,y,z) =  GradedPolynomialRing(QQ, ["x", "y", "z"])
│ (Multivariate Polynomial Ring in x, y, z over Rational Field graded by
│   x -> [1]
│   y -> [1]
│   z -> [1], MPolyElem_dec{fmpq, fmpq_mpoly}[x, y, z])
│ 
│ julia> V = [x,y,x^2+y^2]
│ 3-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
│  x
│  y
│  x^2 + y^2
│ 
│ julia> minimal_subalgebra_generators(V)
│ 2-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
│  y
│  x
│ ```
│ 
│ Subexpression:
│ 
│ minimal_subalgebra_generators(V)
│ 
│ Evaluated output:
│ 
│ 2-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
│  x
│  y
│ 
│ Expected output:
│ 
│ 2-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
│  y
│  x
│ 
│   diff =
│    2-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
│     y
│     xx
│     y
└ @ Documenter.DocTests ~/.julia/packages/Documenter/R2HVS/src/DocTests.jl:385
ERROR: LoadError: `makedocs` encountered a doctest error. Terminating build

Reply:
My code changes also changed the output to something equivalent but differ, hence the failure